### PR TITLE
Add draft for a transformers.PreTrainedModel for MSA Pairformer

### PR DESCRIPTION
**Pull request by mauricebrenner**
Add MsaPairformerPreTrainedModel and MsaPairformer as PreTrainedModel classes Add "fast" tokenizer for MsaPairformer

Move mask creation into MsaPairformer (creates row, column and pairwise mask from MSA attention mask)

Add MsaPairformerEncoder (CoreModule), MsaPairformerContactHead (LogisticRegressionContactHead) and MsaPairformerLmHead (LMHead) nn.Modules to adhere to transformers style of having all modules within one file See e.g.: https://github.com/huggingface/transformers/blob/6f259bc83e518c281877cfc90efe61bf8a79bba0/src/transformers/models/esm/modeling_esm.py

Add new MsaPairformerEmbeddings module for creation of initial representations Move one hot encoding into MsaPairformerEmbeddings

Add push_safetensors.py to copy weights from MSAPairformer (PyTorch) to MsaPairformer (transformers) and push weights to HuggingFace Hub

Add "transformers>=4.50.0" dependency in pyproject.toml Add macOS, JetBrains and Python .gitignore template from https://github.com/github/gitignore

### CODE REVIEW

**Comment by mauricebrenner** [comment]
HuggingFace PR: https://huggingface.co/yoakiyama/MSA-Pairformer/discussions/1

Really keen to use this model with other HuggingFace libraries like PEFT and TRL :)

I hope this can be integrated into transformers at some point, but I would be surprised if they add the einx, einops dependencies etc. for this (really like the jaxtyping + einstein notation combination you use btw 👍)
